src/snarky.d.ts

```diff
@@ -51,489 +53,806 @@ declare interface ProvablePure<T> extends Provable<T> {
    */
-  assertLessThanOrEqual(
-    y: Field | number | string | boolean,
-    message?: string
-  ): void;
+  greaterThan(value: Field | number | string | boolean): Bool;
+  
   /**
+   * Check if this {@link Field} is greater than or equal another "field-like" value.
+   * Returns a {@link Bool}, which is a provable type and can be used to prove the validity of this statement.
    *
-   * Assert that this {@link Field} is greater than another Field-like value.
-   *
+   * @example
    * ```ts
-   * Field(1).assertGt(0);
+   * Field(3).greaterThanOrEqual(3).assertEquals(Bool(true));
    * ```
-   *
-   */
-  assertGreaterThan(
-    y: Field | number | string | boolean,
-    message?: string
-  ): void;
-  /**
-   *
-   * Assert that this {@link Field} is greater than or equal to another Field-like value.
-   *
+   * 
+   * **Warning**: As this method compares the bigint value of a {@link Field}, it can result in unexpected behaviour when used with negative inputs or modular division.
+   * 
+   * @example
    * ```ts
-   * Field(1).assertGte(0);
+   * Field(1).div(Field(2)).greaterThanOrEqual(Field(1).div(Field(3))).assertEquals(Bool(true)); // This code will throw an error
    * ```
-   *
+   * 
+   * @param value - the "field-like" value to compare with this {@link Field}.
+   * 
+   * @return A {@link Bool} representing if this {@link Field} is greater than or equal another "field-like" value.
    */
-  assertGreaterThanOrEqual(
-    y: Field | number | string | boolean,
-    message?: string
-  ): void;
+  greaterThanOrEqual(value: Field | number | string | boolean): Bool;
 
   /**
-   * @deprecated Deprecated - use {@link lessThan} instead
-   *
-   * Check if this {@link Field} is lower than another Field-like value.
-   * Returns a {@link Bool}.
-   *
-   * ```ts
-   * Field(2).lt(3); // Bool(true)
-   * ```
+   * Assert that this {@link Field} is less than another "field-like" value.
+   * It is equivalent to `Field(...).lessThan(...).assertEquals(Bool(true))`.
+   * Please see {@link Field.lessThan} for more details.
+   * 
+   * **Important: If an assertion fails, the code throws an error.
+   * 
+   * @param value - the "field-like" value to compare & assert with this {@link Field}.
+   * @param message? - a string error message to print if the assertion fails, optional.
    */
-  lt(y: Field | number | string | boolean): Bool;
+  assertLessThan(value: Field | number | string | boolean, message?: string): void;
+  
   /**
-   * @deprecated Deprecated - use {@link lessThanOrEqual} instead
-   *
-   * Check if this {@link Field} is lower than or equal to another Field-like value.
-   * Returns a {@link Bool}.
-   *
-   * ```ts
-   * Field(2).lte(3); // Bool(true)
-   * ```
+   * Assert that this {@link Field} is less than or equal another "field-like" value.
+   * It is equivalent to `Field(...).lessThanOrEqual(...).assertEquals(Bool(true))`.
```

Comment:
what is the subject of "It" here? let's state the subject 

maybe we mean "This {@link Field} is equivalent..." ?